src/components/datepicker/dateLocaleProvider.js

```diff
@@ -53,7 +53,8 @@
    *     };
    *
    *     $mdDateLocaleProvider.formatDate = function(date) {
-   *       return moment(date).format('L');
+   *       var m = moment(date);
+   *       return m.isValid() ? m.format('L') : date;
```

Comment:
It should still return a string when the date is not valid; probably the empty string.